src/chainstate/stacks/index/marf.rs

```diff
@@ -606,6 +606,36 @@ impl MARF {
         self.storage.chain_tips()
     }
 
+    /// Check if a block can open successfully, i.e.,
+    ///   it's a known block, the storage system isn't issueing IOErrors, _and_ it's in the same fork
+    ///   as the current block
+    /// The MARF _must_ be open to a valid block for this check to be evaluated.
+    pub fn check_block_hash(&mut self, bhh: &BlockHeaderHash) -> Result<(), Error> {
+        let cur_block_hash = self.storage.get_cur_block();
+        let cur_block_rw = self.storage.readwrite();
+
+        let cur_block_fork_ptr = self.storage.fork_table.get_fork_ptr(&cur_block_hash)
+            .unwrap_or_else(|e| match e {
+                Error::NotFoundError => panic!("Fork table does not contain entry for the current MARF block {:?}", cur_block_hash),
+                e => panic!("Unexpected fork table error {}", e)
+            });
+
+        let bhh_fork_ptr = self.storage.fork_table.get_fork_ptr(bhh)?;
+
+        if bhh_fork_ptr.fork_id != cur_block_fork_ptr.fork_id {
```

Comment:
This isn't an error.  All you want to be true is that `bhh` is the hash of an ancestor of the current block (identified by `cur_block_hash`).  `bhh` can live in a different fork column, which is what `fork_id` identifies.

What you want to do is first calculate the block height of `bhh` (call it `bhh_block_height`), and then call `self.storage.fork_table.walk_back(&cur_block_hash, bhh_block_height)` to get the ancestor block hash.  If that block hash is _not_ `bhh`, then `bhh` is not an ancestor of `cur_block_hash` and the method should fail.

Reply:
I see -- will update.